anap1003/assignment2/TreeNode.java

```diff
@@ -0,0 +1,47 @@
+package assignment2;
+
+public class TreeNode<T> {
+    private T key;
+    private TreeNode parent, leftChild, rightChild;
+
+    public TreeNode(T key) {
+        this.key = key;
+    }
+
+    public TreeNode(T key, TreeNode parent) {
+        this.key = key;
+        this.parent = parent;
+    }
+
+    public TreeNode getParent() {
+        return parent;
+    }
+
+    public T getKey() {
+        return key;
+    }
+
+    public TreeNode getLeftChild() {
+        return leftChild;
+    }
+
+    public TreeNode getRightChild() {
+        return rightChild;
+    }
+
+    public void setLeftChild(TreeNode leftChild) {
+        this.leftChild = leftChild;

```

Comment:
You are setting the child independently of setting the parent; this could result in an incorrect Tree if the caller is not careful. Having them public allows a caller to break the tree. How could you protect this functionality?